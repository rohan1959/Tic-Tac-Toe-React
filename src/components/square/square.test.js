import Square from './square';
import { shallow } from 'enzyme';


describe('test square component', () => {
  test('render square with button with value as passes by prop initially', () => {
    const wrapper = shallow(<Square value='X' />);
    expect(wrapper).toMatchSnapshot();
  });

  test('button to have on click property', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<Square onClick={mockCallBack} />);
    wrapper.find('button').simulate('click');
    expect(mockCallBack).toHaveBeenCalled();
  })
});

