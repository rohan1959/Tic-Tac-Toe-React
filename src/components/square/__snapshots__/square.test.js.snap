// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test square component render square with button with value as passes by prop initially 1`] = `ShallowWrapper {}`;
